src/services/apiService.ts

```diff
@@ -0,0 +1,105 @@
+import type { RestAgentModules } from '../cliAgent'
+
+import { Agent } from '@credo-ts/core'
+import { injectable } from 'tsyringe'
+@injectable()
+export class ApiService {
+  private fetch: typeof globalThis.fetch
+
+  public constructor(agent: Agent<RestAgentModules>) {
+    this.fetch = agent.config.agentDependencies.fetch
+  }
+
+  public async postRequest<T, R>(url: string, payload: T, token: string): Promise<R> {
+    const headers = {
+      Accept: '*/*',
+      Authorization: `Bearer ${token}`,
+      'Content-Type': 'application/json',
+    }
+
+    try {
+      const response = await this.fetch(url, {
+        method: 'POST',
+        headers,
+        body: JSON.stringify(payload),
+      })
+
+      if (!response.ok) {
+        throw new Error(`Failed to make POST request. Status code: ${response.status}`)
+      }
+
+      return (await response.json()) as R
+    } catch (error) {
+      throw new Error(`Error making POST request: ${error}`)
+    }
+  }
+
+  public async getRequest(url: string, token?: string): Promise<any> {
```

Comment:
We can have an option to set the return type here
```suggestion
  public async getRequest<R = any>(url: string, token?: string): Promise<R> {
```